kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/FilterReconciler.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.ControllerConfiguration;
+import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.filter.api.v1alpha1.RecordEncryption;
+
+@ControllerConfiguration
+public class FilterReconciler implements // EventSourceInitializer<RecordEncryption>,
+        Reconciler<RecordEncryption> {
```

Comment:
so is the operator going to need knowledge of each Filter type in code to reconcile them? Or will we be able to build something more generic?

Reply:
This class should not have been added. It's not used and I'll remove it when I next push some changes to this PR.

To answer your question: I'm aiming for the operator not have have a Java-level dependency on any filter class. The operator will have to load knowledge from _somewhere_ of the filter APIs and how they're associated with the filter java class names (represented as `String` within the operator process). 

Candidates for _somewhere_ include: 
* Some YAML baked into the container image, 
* Some YAML in a ConfigMap, that's volume-mounted into the operator container,
* Some `OperatorConfig` CR which the operator gets from the kube API when starting up